frontend/src/scenes/dashboard/dashboardLogic.tsx

```diff
@@ -263,11 +268,11 @@ export const dashboardLogic = kea<dashboardLogicType>({
         ],
         dashboard: [
             () => [dashboardsModel.selectors.sharedDashboards, dashboardsModel.selectors.dashboards],
-            (sharedDashboards, dashboards) => {
-                if (sharedDashboards && !!sharedDashboards[props.id]) {
+            (sharedDashboards, dashboards): DashboardType | null => {
+                if (sharedDashboards && props.id && !!sharedDashboards[props.id]) {
                     return sharedDashboards[props.id]
                 }
-                return dashboards.find((d) => d.id === props.id)
+                return dashboards.find((d) => d.id === props.id) ?? null
```

Comment:
We don't check that props.id is defined before this check. So potentially if there is a dashboard with an undefined id  and props.id is undefined then this will match.

Feels like the protection should be that a dashboard cannot have an undefined id over a check here though

Reply:
This is a nice catch and I think this hotfix should get the job done for now. Agreed that we could change later how dashboardLogic's are consumed throughout the app.